src/update/UpdateManager.cpp

```diff
@@ -130,6 +130,42 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
 
                 if (code == wxID_YES)
                 {
+                    const audacity::network_manager::Request downloadRequest(mVersionPatch.download.ToStdString());
+                    auto downloadResponse = audacity::network_manager::NetworkManager::GetInstance().doGet(downloadRequest);
+
+                    downloadResponse->setRequestFinishedCallback([downloadResponse, this](audacity::network_manager::IResponse*) {
+                        mProgressDialog.reset();
+                        }
+                    );
+
+                    downloadResponse->setDownloadProgressCallback(
+                        [downloadResponse, this](int64_t current, int64_t expected) {
+
+                            static std::once_flag flag;
+                            std::call_once(flag, [this] {
```

Comment:
Is call_once really right?  Or should you do this each time?  See elsewhere about lifetime of the progress dialog.